### PR TITLE
Add footer layout component

### PR DESCRIPTION
**Pull request by GeorgDV**
Changes from #26 (not to close yet)

---

This adds minimal footer layout component.

Updated visual after (with bloated content):
updatedfooter

On empty page:
updatedfooter empty page

### CODE REVIEW

File: fe/src/pages/About/About.tsx [hunk 1]
```diff
@@ -1,9 +1,11 @@
+import { Box } from '@chakra-ui/react';
+
 function About(): JSX.Element {
   // @TODO - Add some actual content here.
   return (
-    <div>
+    <Box h="100vh">
       <h1>About Page</h1>
-    </div>
+    </Box>
```
Comment (GeorgDV) [inline fe/src/pages/About/About.tsx:8]: Temporary visual change for "empty" pages, just to make current footer position lower.

File: fe/src/pages/About/About.tsx [hunk 2]
```diff
@@ -1,9 +1,11 @@
+import { Box } from '@chakra-ui/react';
+
 function About(): JSX.Element {
   // @TODO - Add some actual content here.
   return (
-    <div>
+    <Box h="100vh">
```
Comment (GeorgDV) [inline fe/src/pages/About/About.tsx:6]: Temporary visual change to make footer position lower for "empty" pages.

File: fe/src/pages/Error/Error.tsx [hunk 3]
```diff
@@ -1,12 +1,13 @@
+import { Box } from '@chakra-ui/react';
 import { Link } from 'react-router-dom';
 
 function Error(): JSX.Element {
   // @TODO - Add some actual content here.
   return (
-    <div>
+    <Box h="100vh">
```
Comment (Orzelius) [inline fe/src/pages/Error/Error.tsx:7]: I think it's a bad idea to relay on other elements to be 100vh in order for the footer to stay on the bottom. Maybe create a container element in the `/App.tsx` and make that 100vh (also might as well assign a smaller width as well) and then place the router paths in it. This way anything that will be added in the future will stay in the container.

Or just float the footer to the bottom

**Review by Orzelius — COMMENTED** [review]
I'll leave the positioning decision up to you

File: fe/src/components/layouts/Footer/Footer.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      <Container
+        as={Stack}
+        maxW="8xl"
+        py={4}
+        direction={{ base: 'column', md: 'row' }}
+        spacing={4}
+        justify={{ base: 'center', md: 'space-between' }}
+        align={{ base: 'center', md: 'center' }}
+      >
+        <Text>© 2021 Tallinn Polytechnic School. All rights reserved</Text>
```
Comment (Kasparsu) [inline fe/src/components/layouts/Footer/Footer.tsx:24]: maybe add link to TPT website (gotta promote the school somehow)

Comment (GeorgDV) [reply fe/src/components/layouts/Footer/Footer.tsx:24]: Sure thing.

File: fe/src/pages/Error/Error.tsx (hunk 3 above)
Comment (Kasparsu) [reply fe/src/pages/Error/Error.tsx:7]: https://jsfiddle.net/wnLT7/6/

File: fe/src/pages/About/About.tsx (hunk 2 above)
Comment (Kasparsu) [reply fe/src/pages/About/About.tsx:6]: https://jsfiddle.net/wnLT7/6/

Comment (Orzelius) [reply fe/src/pages/About/About.tsx:6]: With Kaspars approach you also have to add padding to the wrapper so the footer wouldn't cover it up when you have longer content.

**Review by tomimarkus991 — APPROVED** [review]
Looks good. Well done!

**Comment by GeorgDV** [comment]
> Looks good. Well done!

@tomimarkus991 Just little early on the review, was still working on final changes. 😄

File: fe/src/pages/About/About.tsx (hunk 2 above)
Comment (GeorgDV) [reply fe/src/pages/About/About.tsx]: Went with @Kasparsu approach, check above for updated visual picture.

**Review by Orzelius — APPROVED** [review]
image
The small external link icon is a bit wonky but other than that all good

**Review by Kasparsu — APPROVED** [review]

File: fe/src/App.tsx [hunk 5]
```diff
@@ -2,19 +2,20 @@ import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
...
     <>
-      <Header />
-      <Router>
-        <Switch>
-          <Route exact path="/" component={Home} />
-          <Route path="/about" component={About} />
-          <Route component={Error} />
-        </Switch>
-      </Router>
+      <Default>
```
Comment (Kasparsu) [inline fe/src/App.tsx:10]: currently this is fine but in future pr need to think how to implement per page switching.

**Pull request merged**